### PR TITLE
Test data generation

### PR DESCRIPTION
**Pull request by JMBattista**
Support generating test sequences and test reads from those sequences.

### CODE REVIEW

File: sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    public void testCreateNoEncodingScheme() {
+        SequenceGenerator.builder()
+                .lengthDistribution(() -> 10)
+                .basePairDistribution(defaultDistribution)
+                .build();
+    }
+
+
+    /**
+     * Attempting to create without an encoding scheme fails
```
Comment (ctdavids) [inline sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java]: Comment should refer to missing length not scheme

File: sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        SequenceGenerator.builder()
+                .encodingScheme(AcceptUnknownDnaEncodingScheme.instance)
+                .basePairDistribution(defaultDistribution)
+                .build();
+    }
+
+
+    /**
+     * Attempting to create without an encoding scheme fails
+     */
```
Comment (ctdavids) [inline sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java]: Comment should refer to missing base pair distribution not encoding scheme.

File: sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                .build();
+
+        assertThat(generator.sample().length(), is(length));
+    }
+
+
+    /**
+     * Test that we can generate a sequence from an {@link EnumeratedDistribution} containing only one element
+     */
+    @Test
```
Comment (ctdavids) [inline sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java]: As with previous A/T tests this one and the subsequent one share nearly all code differing only on the base pair and could be generalized if desired.

File: sequence.generator/src/main/java/com/vitreoussoftware/bioinformatics/sequence/generator/distribution/EnumeratedDistribution.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+ * If you have continues values you may want to look at {@see ContinuousDistribution}
+ *
+ * Created by John on 10/16/2016.
+ */
+public class EnumeratedDistribution<T> implements Distribution<T> {
+    private final org.apache.commons.math3.distribution.EnumeratedDistribution<T> distribution;
+
+    /**
+     * Create an instance of {@link EnumeratedDistribution} that can be used to generate outputs
+     * @param probabilities Probabilitis for the elements of the distribution
```
Comment (JMBattista) [inline sequence.generator/src/main/java/com/vitreoussoftware/bioinformatics/sequence/generator/distribution/EnumeratedDistribution.java]: fixed

File: sequence.generator/src/main/java/com/vitreoussoftware/bioinformatics/sequence/generator/distribution/EnumeratedDistribution.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+     * @param normalizeProbabilities EnumeratedDistribution values for BP values we want to accept
+     *
+     * @throws NullPointerException If the encoding scheme is not set
+     * @throws IllegalArgumentException If the probabilities values are not set
+     */
+    EnumeratedDistribution(
```
Comment (JMBattista) [inline sequence.generator/src/main/java/com/vitreoussoftware/bioinformatics/sequence/generator/distribution/EnumeratedDistribution.java]: Added

File: sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/ReadGeneratorTest.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+ * Test the {@link ReadGenerator}
+ *
+ * Created by John on 10/27/2016.
+ */
+public class ReadGeneratorTest {
+
+    /**
+     * Test that we can sample length 1 reads from a length 1 sequence
+     */
+    @Test
```
Comment (JMBattista) [inline sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/ReadGeneratorTest.java]: Switched to thoeries

File: sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java (hunk 1 above)
Comment (JMBattista) [inline sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java]: fixed

File: sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java (hunk 3 above)
Comment (JMBattista) [inline sequence.generator/src/test/java/com/vitreoussoftware/bioinformatics/sequence/generator/SequenceGeneratorTest.java]: Switched to theories

**Pull request merged**